### PR TITLE
Improve StatsMakie documentation

### PR DESCRIPTION
**Pull request by asinghvi17**
See also https://github.com/JuliaPlots/Makie.jl/issues/307#issuecomment-477246493.  StatsMakie is underdocumented, and some examples of the undocumented use exist in MakieThemes and elsewhere.  These should be added to the example gallery and the StatsMakie page.

Todo list:

- [x] Examples showcasing analyses
- [x] Examples showcasing plot recipes (as in MakieThemes)
- [x] Description of analyses
- [x] Description of what is available

### CODE REVIEW

**Comment by asinghvi17** [comment]
Next step is to add examples - the MakieThemes ones to start (using RDatasets instead of TSV's bundled with MakieThemes).

**Comment by asinghvi17** [comment]
Since the `www` dataset is not present in RDatasets, a replacement dataset will have to be found.

**Comment by SimonDanisch** [comment]
In this repo, we could also ship data, if it's ok with the license ;)

**Comment by asinghvi17** [comment]
I think these are in a good spot, if corrplot is not broken then that's about all I can add...I assume the dropdown widgets are unimplemented so the gui for a dataframe can't be used?

**Comment by SimonDanisch** [comment]
Awesome :) @piever if you have time can you take a quick look? If not, I guess I'm just going ahead to merge this!

File: docs/src/statsmakie.md [hunk 1]
```diff
@@ -120,3 +120,55 @@ And everything else works as usual:
...
+
+There are multiple options with which to analyze your data before plotting it.  These are:
+
+- density (kernel density estimation, 1D or 2D)
+- histogram (1D, 2D or even 3D!)
+- frequency (count occurrences of discrete variables, 1 or 2D)
+- linear (linear regression)
+- smooth (LOESS regression)
+
+To use these analyses, one can simply write something like `plot(density, x, y)`.  One can also pass options to the analysis, as in: `plot(density(bandwidth-0.1), x, y)`, or something analogous for other analyses.
```
Comment (piever) [inline docs/src/statsmakie.md]: Typo: should be `density(bandwidth=0.1)`

File: docs/src/statsmakie.md [hunk 2]
```diff
@@ -120,3 +120,55 @@ And everything else works as usual:
...
+
+## Statistical plot types
+
+One can use box plots and violin plots with the same interface as `StatsPlots`.  
+
+One can create a box plot
+
+@example_database("Box plot")
+
+or a violin plot!
```
Comment (piever) [inline docs/src/statsmakie.md]: Maybe colon instead of exclamation mark?

**Comment by piever** [comment]
Great! I've added two tiny spelling comments, but overall looks very nice!

**Pull request merged**